language/move-prover/bytecode-to-boogie/src/env.rs

```diff
@@ -264,15 +244,68 @@ pub struct ModuleEnv<'env> {
 
 impl<'env> ModuleEnv<'env> {
     /// Returns the index of this module in the global env.
-    pub fn get_module_idx(&self) -> usize {
+    pub fn get_module_idx(&self) -> ModuleIndex {
         self.data.idx
     }
 
-    /// Returns the name of this module.
+    /// Returns the id of this module.
     pub fn get_id(&'env self) -> &'env ModuleId {
         &self.data.id
     }
 
+    /// Adds diagnostic to the environment.
+    pub fn add_diag(&self, diag: Diagnostic) {
+        self.data.diags.borrow_mut().push(diag);
+    }
+
+    /// Returns true if diagnostics have error severity or worse.
+    pub fn has_errors(&self) -> bool {
+        self.data
+            .diags
+            .borrow()
+            .iter()
+            .any(|d| d.severity <= Severity::Error)
```

Comment:
> error severity or worse.

Should this be `>= Severity::Error`, then (based on [this](https://docs.rs/codespan-reporting/0.1.4/codespan_reporting/enum.Severity.html))? I think the code as it is currently written will skip reporting `Severity::Bug`s.

Reply:
Good catch! Ordering is reverse to enum declaration ordering, which is confusing. Done.